[Anvita2110] iP

Anvita2110 opened the pull request:
# Ponder Pika
> Let our advance worrying become advance thinking and planning - Winston Churchill

Ponder Pika lets you manage your tasks without having to keep all of them in your mind.

* Text-based
* Easier commands
* ~~Simple~~ Simpler usability

All you need to do is,

1. Download it from [here](https://github.com/Anvita2110/ip/releases/tag/A-Jar).
2. Run in your CMD.
3. Add your tasks using the commands.
4. Let it manage your tasks for you 😉

And all this for **FREE**

Features:

- [X] Manages Tasks
- [X] Manages Deadlines
- [X] Manages Events
- [ ] Reminders \(Coming soon)

---


If you are a programmer, you can make one of yours too!
The `main` method should look something like this:

```Java
public class Main {
    public static void main(String[] args) {
        Ponder_Pika bot = new Ponder_Pika();
        bot.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Anvita2110 committed:
Change chatbot name

Anvita2110 committed:
Add greeting

Anvita2110 committed:
add echo method

Anvita2110 committed:
add command to list

Anvita2110 committed:
add listing feature

Anvita2110 committed:
modify list feature

Anvita2110 committed:
Improve Logic

Anvita2110 committed:
Complete Level-3

Anvita2110 committed:
Add Todo an Deadline class with implementation

Anvita2110 committed:
Add Event Task

Anvita2110 committed:
Handle missing argument exception

Anvita2110 committed:
Handle index errors for mark and unmark

Anvita2110 committed:
Handle missing time exceptions - Level 5

Anvita2110 committed:
Add delete feature for bot - Level 6

Anvita2110 committed:
Update Task.java and subclasses

Anvita2110 committed:
Add Save/Load feature

Anvita2110 committed:
Merge branch 'branch-Level-7'

Anvita2110 committed:
Add Data and Time feature

Anvita2110 committed:
Merge branch 'branch-Level-8'

Anvita2110 committed:
Add find feature

Anvita2110 committed:
Follow Code Standard

Anvita2110 committed:
Follow coding standards - 1

Anvita2110 committed:
Follow coding standards - 2

Anvita2110 committed:
Follow coding standards - 3

Anvita2110 committed:
Add JavaDoc comments to main files

Anvita2110 committed:
Add JavaDoc comments to remaining files

Anvita2110 committed:
Add JavaDoc comments to remaining files

Anvita2110 committed:
Merge branch 'branch-A-CodingStandard'

Anvita2110 committed:
Solve Merge Conflict

Anvita2110 committed:
Merge Branch-Level-9 in master

Anvita2110 committed:
Add Text-Ui-Testing

Anvita2110 committed:
Add Text-Ui-Testing

Anvita2110 committed:
Add Ui class

Anvita2110 committed:
Abstract out new Classes

Anvita2110 committed:
Abstract out new Classes

Anvita2110 committed:
Merge branch 'branch-A-MoreOOP'

Anvita2110 committed:
Merge branch 'add-gradle-support' of https://github.com/Anvita2110/ip into add-gradle-support

Anvita2110 committed:
Merge branch 'add-gradle-support'

Anvita2110 committed:
Add Gradle to Project

Anvita2110 committed:
Add unit tests for Todo

taniathng commented on src/main/java/Ponder_Pika/Task/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+     * @param description the description of the task
+     * @param deadline the deadline by which the task should be completed
+     */
+    public Deadline(String description, LocalDateTime deadline) {
```
I like the clear javadocs ! keep up the good work!

Anvita2110 replied:
Thank you for the review!

taniathng commented on src/main/java/Ponder_Pika/Main.java:
```diff
@@ -0,0 +1,8 @@
+
+public class Main {
+    public static void main(String[] args) {
+        Ponder_Pika bot = new Ponder_Pika();
```
Is Ponder_Pika in PascalCase?

Anvita2110 replied:
It is not, I have changed it completely. I did my check style after your reviews were given.

taniathng commented on src/main/java/Ponder_Pika/Task/Event.java:
```diff
@@ -0,0 +1,61 @@
+     * @return a string representing the full details of the event
+     */
+    @Override
+    public String saveFullDetails() {
```
Good Job on consistency naming methods in cameCase!

Anvita2110 replied:
Thank you for commenting on this!

taniathng reviewed (commented):
Good job overall! keep up the good work!

Anvita2110 committed:
Use checkstyle for code style variations

Anvita2110 committed:
Add GUI and Launcher

Anvita2110 committed:
Add pictures and add styles to controls

Anvita2110 committed:
Add handle command method

Anvita2110 committed:
Set up GUI from CLI

Anvita2110 committed:
Add MainWindow and DialogBox FXML files

Anvita2110 committed:
Tweak GUI for layout

Anvita2110 committed:
Merge branch 'branch-Level-10'

Anvita2110 committed:
Change DialogBox FXML file

Anvita2110 committed:
Update existing files for functionality

Anvita2110 committed:
Improve code quality for few classes

Critically examined the code and refactored to enhance overall code
quality. This includes:
* Improve readability and maintainability
* Applying coding standards and best practices
* Ensuring consistency across the codebase

Anvita2110 committed:
Add assertions to Parser, TaskList and IoHandler

Pareser, TaskList and IoHandler files do not use assertions

These files can have safer excution with the help of assertions
by validating key assumptions directly in the code which
helps in catching potential issues early.

- Added assertions to verify preconditions in key functions.
- Used assertions to enforce invariants that should be
maintained during execution.

Native assert statements are used instead of JUnit assertions to
directly embed assumptions within the code, making them part of
the runtime checks and ensuring they are always enforced.

Anvita2110 committed:
Add streams in PonderPika file

The current code iterates through tasks using traditional loops and
manual filtering.

Refactor the code to utilize Java Streams for filtering and formatting
tasks.

Using Streams improves readability and conciseness. The code now
leverages functional programming constructs to filter tasks and format
the output in a more expressive manner.

No functional changes are introduced; this is a refactor to enhance code
clarity and maintainability.

Anvita2110 committed:
Merge pull request #1 from Anvita2110/branch-A-CodeQuality

Improve code quality for few classes

Anvita2110 committed:
Fix branch merge conflicts - branch-A-Assertions

Anvita2110 committed:
Fix branch merge conflicts - branch-A-Streams

Anvita2110 committed:
Merge pull request #2 from Anvita2110/branch-A-Streams

Add streams to the code

Anvita2110 committed:
Merge pull request #3 from Anvita2110/branch-A-Assertions

Add assertions to few files

Anvita2110 committed:
Set up Continous Integration

Anvita2110 committed:
Add C-Priority extension for tasks

Anvita2110 committed:
Update C-Priority extension

Anvita2110 committed:
Update BCD Extension due to errors

Anvita2110 committed:
Merge branch 'branch-BCD'

Anvita2110 committed:
Handle different exceptions seperately

Anvita2110 committed:
Introduce more exception handling methods

Anvita2110 committed:
Improve exception handling for parsing and input

The application currently has limited error handling for user input and
environmental issues, which can lead to crashes or undefined behavior.

Better error handling is essential for providing a better
user experience and ensuring application stability.

Comprehensive error handling will be implemented for command parsing,
including checks for format errors, and data consistency.

This approach informs users of their mistakes and improves the
application's resilience to unexpected input and conditions.

Anvita2110 committed:
Change gradle.yml file

Anvita2110 committed:
Change gradle.yml file in branch

Anvita2110 committed:
Merge branch 'branch-MoreErrorHandling'

Anvita2110 committed:
Update files for IO queries

Anvita2110 committed:
Correct bug related to checkstyle error

Anvita2110 committed:
Updated the personality for Ponder Pika

Anvita2110 committed:
Handle errors reported from PR

Anvita2110 committed:
fix checkstyle

Anvita2110 committed:
fix testcase

Anvita2110 committed:
Merge pull request #9 from Anvita2110/branch-A-Personality

Updated the personality for Ponder Pika

Anvita2110 committed:
Add code reuse comments

Anvita2110 committed:
Add Ui.png and README.md file

Anvita2110 committed:
Introduce Junit tests for TaskList class

The TaskList class lacks comprehensive unit tests.

This risks undetected bugs and functionality issues.

Add more JUnit tests to cover all methods in the TaskList class.

This improves the reliability of the TaskList class by
verifying its behavior in a wide range of cases.

Anvita2110 committed:
Updated file to handle PR request

Anvita2110 committed:
Updated file again to handle PR request

Anvita2110 committed:
Updated file to handle PR request-3

Anvita2110 committed:
Merge pull request #10 from Anvita2110/branch-A-MoreTesting

Introduce Junit tests for TaskList class

Anvita2110 committed:
Reference to ChatGPT - commit message template

Anvita2110 committed:
Updated branch-A-MoreGUI

Anvita2110 committed:
Update branch-A-MoreGUI

Anvita2110 committed:
Merge pull request #11 from Anvita2110/branch-A-MoreGUI

Updated branch-A-MoreGUI

Anvita2110 committed:
Updated Ui.png

Anvita2110 committed:
Update Ui.png in docs

Anvita2110 committed:
Fixed Checkstyle

Anvita2110 committed:
Fix checkstyle errors

Anvita2110 committed:
Merge branch 'master' of https://github.com/Anvita2110/ip

Anvita2110 committed:
Merge branch 'master' of https://github.com/Anvita2110/ip

Anvita2110 committed:
Merge branch 'master' of https://github.com/Anvita2110/ip
